proposals/0107-unsaferawpointer.md

```diff
@@ -1477,34 +1477,36 @@ struct UnsafeMutableRawPointer : Strideable, Hashable, _Pointer {
   func bindMemory<T>(to: T.Type, capacity: Int) -> UnsafeMutablePointer<T>
   func assumingMemoryBound<T>(to: T.Type) -> UnsafeMutablePointer<T>
 
-  func initialize<T>(_: T.Type, with: T, count: Int = 1)
+  func initialize<T>(_: T.Type, to: T, count: Int = 1)
     -> UnsafeMutablePointer<T>
-  func initialize<T>(toContiguous: T.Type, atIndex: Int, with: T)
+  func initialize<T>(contiguous: T.Type, at: Int, to: T)
```

Comment:
Yes, binding is idempotent!
There is an important difference between `rawPtr.initializeMemory(as: T.self, to: T())` and `ptrToT.initialize(to: Pointee())` because the raw initialize can be called to `rebind` to a different type. The later has a precondition that the memory is already bound to `Pointee`.

That difference is somewhat signaled by the `as` label. However, I think it's also good to give the `raw` version a different name to clearly distinguish them. It then reads nicely as "initialize memory as `T`" because changing the memory's bound type is as important to this operation as changing the initialized state.

I do *not* want to rename all of the `UnsafePointer<T>` APIs with the `Memory` suffix, mainly because it would be excessive, cumbersome, and wouldn't solve any real usability problem:
- initialize
- moveInitialize
- assign
- moveAssign

I think "memory" is the obviously implied target of this API family.